File: scripts/_patch_ko_weather.py
# -*- coding: utf-8 -*-
path = "src/i18n/locales/ko.json"
with open(path, "r", encoding="utf-8") as f:
    text = f.read()
old = '    "dataSync": "데이터 동기화",\n    "dataReview":'
new = '    "dataSync": "데이터 동기화",\n    "weatherRecords": "기상 기록",\n    "dataReview":'
if old not in text:
    raise SystemExit("pattern not found")
if '"weatherRecords"' in text[:2000]:
    print("skip: weatherRecords may already exist near top")
else:
    text = text.replace(old, new, 1)
    with open(path, "w", encoding="utf-8", newline="\n") as f:
        f.write(text)
    print("patched")

File: scripts/patch-ko-weather.mjs
import fs from "fs";

const p = "src/i18n/locales/ko.json";
let t = fs.readFileSync(p, "utf8");
const oldStr =
  '    "dataSync": "\uB370\uC774\uD130 \uB3D9\uAE30\uD654",\n    "dataReview":';
const newStr =
  '    "dataSync": "\uB370\uC774\uD130 \uB3D9\uAE30\uD654",\n    "weatherRecords": "\uAE30\uC0C1 \uAE30\uB85D",\n    "dataReview":';
if (!t.includes(oldStr)) {
  console.error("pattern not found");
  process.exit(1);
}
if (t.slice(0, 2500).includes('"weatherRecords"')) {
  console.log("skip: already present in sidebar");
} else {
  t = t.replace(oldStr, newStr);
  fs.writeFileSync(p, t, "utf8");
  console.log("patched");
}
